apps/ringtones/js/share.js

```diff
@@ -228,12 +228,9 @@ function handleShare(data, callback) {
         // Show a notification indicating success, and then close it immediately
         // so it doesn't stink up the notifications tray! XXX: This UX isn't
         // great; we should turn this into a transient notification when we can.
-        new Notification(
-          songtitle,
-          {body: navigator.mozL10n.get(
-            details.setAsDefault ? 'set-default-tone' : 'created-tone'
-          )}
-        ).close();
+        NotificationHelper.send(songtitle, {
+          bodyL10n: details.setAsDefault ? 'set-default-tone' : 'created-tone'
+        }).close();
```

Comment:
the close will not work here. NotificationHelper.send returns a Promise that returns the notification.

You need to take this Promise and use promise.then() and inside it close the notification.